src/mapwindow.jl

```diff
@@ -116,6 +123,89 @@ function _mapwindow{T,N}(f,
     out
 end
 
+
+
+# This is a implementation of Fast 2D median filter
+# http://ieeexplore.ieee.org/document/1163188/
+function median_filter{T,N}(f,
+                         img::AbstractArray{T,N},
+                         window::Indices{N},
+                         border::BorderSpecAny,
+                         shape=default_shape(f);
+                         callmode=:copy!)
+    inds = indices(img)
```

Comment:
Since the original algorithm is faster for smaller window sizes, you could insert something like this:
```julia
if prod(map(length, window)) < 81
    return _mapwindow!(...)
end
```
and keep the original implementation as well.

*However*, this makes testing a little harder: you'd like to check, for example, that the two algorithms give the same result as one another, and that means you need to have a way of calling each one separately.

For that reason, I recommend something like the following:
```julia
# Let's change the syntax of `replace_function`. First, fallbacks in case anyone has already made a local extension.
replace_function(f, window) = replace_function(f)
replace_function(f) = f
# Now the one you care about
function replace_function(::typeof(median!), window)
    if prod(map(length, window)) < 81
        return median_direct!
    end
    median_histogram
end
```
Then change all callers of `replace_function(f)` to `replace_function(f, window)`, define
```julia
function median_direct!(v::AbstractVector)
    inds = indices(v,1)
    Base.middle(Base.select!(v, (first(inds)+last(inds))÷2, Base.Order.ForwardOrdering()))
end
```
and define `median_histogram` to be your definition. Then you can call this method
```julia
function _mapwindow{T,N}(f::typeof(median_histogram),
```

Finally, in tests you can do this:
```julia
med1 = ImageFiltering._mapwindow(ImageFiltering.median_direct!, args...)
med2 = ImageFiltering._mapwindow(ImageFiltering.median_histogram, args...)
@test med1 == med2
```